src/libawkward/array/IndexedArray.cpp

```diff
@@ -2303,6 +2323,7 @@ namespace awkward {
       util::handle_error(err, classname(), identities_.get());
 
       ContentPtr next = content_.get()->carry(nextcarry);
+      // shouldn't we carry starts and stops here as well?
       return next.get()->getitem_next_jagged(slicestarts,
                                              slicestops,
                                              slicecontent,
```

Comment:
It doesn't look to me like anything special is needed here. The non-option-type case is very different from the option-type case, and the feature you're adding only affects option-type cases.

From just staring at it, though, I can't tell if what's here is right. If this section has coverage, then it's very likely right.

Reply:
It has coverage and I don't think there's any issue. I realized what I was doing with the starts and stops isn't really _carry_ but more like reduce